transport/http/outbound.go

```diff
@@ -445,6 +400,118 @@ func getYARPCErrorFromResponse(response *http.Response, bothResponseError bool)
 	)
 }
 
+// RoundTrip implements the http.RoundTripper interface, making a YARPC HTTP outbound suitable as a
+// Transport when constructing an HTTP Client. An HTTP client is suitable only for relative paths to
+// a single outbound service. The HTTP outbound overrides the host:port portion of the URL of the
+// provided request.
+//
+// Sample usage:
+//
+//  client := http.Client{Transport: outbound}
+//
+// Thereafter use the Golang standard library HTTP to send requests with this client.
+//
+//  ctx := context.Background()
+//  ctx, cancel := context.WithTimeout(ctx, time.Second)
+//  defer cancel()
+//  req := http.NewRequest("GET", "http://example.com/", nil)
+//  req = req.WithContext(ctx)
+//  res, err := client.Do(req)
+//
+// All requests must have a deadline on the context.
+// The peer chooser for raw HTTP requests will receive a blank YARPC transport.Request, which is
+// sufficient for load balancers like peer/pendingheap (fewest-pending-requests) and peer/roundrobin
+// (round-robin).
+func (o *Outbound) RoundTrip(hreq *http.Request) (*http.Response, error) {
+	return o.roundTrip(hreq, nil /* treq */, time.Now())
+}
+
+func (o *Outbound) roundTrip(hreq *http.Request, treq *transport.Request, start time.Time) (*http.Response, error) {
+	ctx := hreq.Context()
+
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return nil, yarpcerrors.Newf(
+			yarpcerrors.CodeInvalidArgument,
+			"missing context deadline")
+	}
+	ttl := deadline.Sub(start)
+
+	// When sending requests through the RoundTrip method, we construct the
+	// transport request from the HTTP headers as if it were an inbound
+	// request.
+	// The API for setting transport metadata for an outbound request when
+	// using the go stdlib HTTP client is to use headers as the YAPRC HTTP
+	// transport header conventions.
+	if treq == nil {
+		treq = &transport.Request{
```

Comment:
If we don't set a body here, middleware that modifies the body will fail right?

Reply:
Ah I see, the middleware is only applied with the full dispatcher etc is used, when only using the outbound as a HTTP client, we can't apply middleware.